lib/xenops_server.ml

```diff
@@ -58,7 +60,7 @@ let ignore_exception msg f x =
 let filter_prefix prefix xs =
   List.filter_map
     (fun x ->
-       if String.startswith prefix x
+       if Stdext.Xstringext.String.startswith prefix x
```

Comment:
You can probably simplify this using Astring. We had started replacing stringext with a string with the ideA of getting rid of id two years ago

Reply:
Do you know if xapi's finally can be replaced cleanly with Fun.protect as well?

Reply:
If you can move to a recent compiler it should be good to go. Be a bit careful as they behave a bit differently: xapi is a catch-all while protect is a bit more sophisticate especially in regards to asyncronous exceptions.

Reply:
We could probably replace the implementation in stdext with a wrapper around Fun.protect matching the semantics we used to have before (IIRC we just need to add a log-and-ignore-exceptions around the finally call, because we care about preserving the original exception, and check what happens with stacktraces).